applications/asset_tracker_v2/src/modules/data_module.c

```diff
@@ -855,10 +855,15 @@ static void data_send_work_fn(struct k_work *work)
 static void requested_data_status_set(enum app_module_data_type data_type)
 {
 	if (!k_work_delayable_is_pending(&data_send_work)) {
-		/* If the data_send_work is not pending it means that the module has already
-		 * triggered an data encode/send.
-		 */
-		LOG_DBG("Data already encoded and sent, abort.");
+		if (data_type == APP_DATA_LOCATION) {
+			/* We may get location related data when a fallback to next method occurs */
+			data_send_work_fn(NULL);
+		} else {
```

Comment:
Allowing location data sending after the first set of data is sent is for the fallbacks. I guess this might be a bug in current implementation too. This adds data transfer between samplings so @simensrostad please comment.

Reply:
What do you mean by data transfer between samplings? Could you elaborate a bit on that

Reply:
I mean there is an additional cloud communication somewhere in the middle of the sampling interval.
Generally the idea in ATv2 is to start sampling, collect various data, send all the data to cloud, and after the sampling interval, do the same thing again. In the middle of the sampling interval, ATv2 is not supposed to send data to cloud.

With this change, we are adding an additional data transfer in the middle of the sampling interval because we first send some location related data where location resolution fails. Then we do fallback in Location library and to get the resolution from this location retrieval into the cloud, we need to transmit that set of data.

As I said, there probably should be something like this irrespective as right now I'm not sure location fallbacks are handled properly. whenever GNSS is not the first priority method.

Reply:
I see, I guess this is fine and a penalty we have to deal with when not being able to resolve the first method.